dvc/command/stages.py

```diff
@@ -0,0 +1,306 @@
+        text.append(display_name)
+    text.truncate(MAX_TEXT_LENGTH, overflow="ellipsis")
+    return text
+
+
+def prepare_stage_name(stage: "Stage", link=False, use_name=False) -> Text:
+    title = Text(overflow="fold")
+    address = stage.addressing
+
+    linked = Style(link="file://" + os.path.abspath(stage.relpath))
+    file_style = Style(color=DVC_BLUE, bold=True, underline=False)
+    name_style = None
+    if link:
+        # we show a link to the dvc.yaml file in the terminal for the stages
+        # need to hover over their name
+        name_style = linked
+        file_style += linked
+
+    if isinstance(stage, PipelineStage):
+        from_pwd = stage.name == address
+        if from_pwd or use_name:
+            title.append(stage.name, style=name_style)
+        else:
+            title.append(stage.relpath, style=file_style)
+            title.append(f":{stage.name}")
+    else:
+        title.append(address, style=file_style)
+    return title
+
+
+def prepare_description(stage: Stage) -> Union[Markdown, Text]:
+    style = "blue"
+    if stage.desc:
+        # not sure if this was ever intended, but markdown looks nice
+        return Markdown(stage.desc, style=style)
+
+    description = generate_description(stage)
+    text = Text(description.strip(), style=style)
+    text.truncate(MAX_TEXT_LENGTH, overflow="ellipsis")
+    return text
+
+
+def list_layout(stages: Iterable[Stage], graph: "nx.DiGraph" = None) -> None:
+    """ Displays stages in list layout using rich """
+
+    LAYOUT_WIDTH = 80
+    LEFT_PAD = (0, 0, 0, 4)
+
+    @render_group()
+    def render_stage(stage: "Stage", idx: int):
+        """Yields renderables for a single stage."""
+        if idx:
+            # separator
+            yield Rule(style=DVC_LIGHT_ORANGE)
+            yield ""
+
+        # title
+        title_table = Table.grid(padding=(0, 1), expand=True)
+        title = prepare_stage_name(stage, link=True)
+
+        # basic info at the right side of the table
+        info = get_info(stage)
+        info_text = prepare_info(info)
+        title_table.add_row(title, info_text)
+
+        # pushing all columns except the first one with title, to the right
+        for idx, column in enumerate(title_table.columns):
+            if not idx:
+                continue
+            column.no_wrap = True
+            column.justify = "right"
+        yield title_table
+        yield ""
+
+        # Table of ancestor nodes
+        nodes_table = Table.grid(padding=(0, 1), expand=True)
+        nodes_text = prepare_graph_text(stage, graph)
+        if nodes_text:
+            nodes_table.add_row(Padding(nodes_text, LEFT_PAD))
+            yield nodes_table
+            yield ""
+
+        desc = prepare_description(stage)
+        yield Padding(desc, LEFT_PAD)
+
+    def column(renderable):
+        """Constrain width and align to center to create a column."""
+        return Align.center(renderable, width=LAYOUT_WIDTH, pad=False)
```

Comment:
@pmrowla, could you please try changing this `center` function to `left` and see? 
Do you prefer it that way? How does it look? I have no strong opinion on the centering.

Reply:
I think it looks better with it `left` aligned.

![Screen Shot 2021-01-05 at 5 33 09 PM](https://user-images.githubusercontent.com/651988/103623999-42315380-4f7c-11eb-954c-89c6ae5d59db.png)

Reply:
@pmrowla, done. Thanks for the suggestion.